src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -203,8 +223,12 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                         PropertyCache = cache;
                     }
                     break;
-                case ClassType.Enumerable:
                 case ClassType.Dictionary:
+                    {
+                        KeyType = converter.KeyType;
```

Comment:
Can we just add this line to the existing case? It would be null for `ClassType.Enumerable` (maybe add a comment).